### DISCUSSION TITLE
(Wasm) Web Authentication Broker Redirect Uri

### DISCUSSION THREAD

**Discussion by BertrandDeSaintBezier** [unoplatform/uno#13903] [Q&A] [answered]
I'm trying to setup authentication with an external identity prodiver. What would the redirect uri be for something that isn't situated at the Root ? For example : `https://xxx:7000/xxx/index.html`. The broker currently returns `http://xxx:7000/authentication-callback?code=yyy`. We tried manually calling it with  `http://xxx:7000/xxx/authentication-callback?code=yyy`, which didn't work. 

Anything that we are missing here ? 

Thanks !

**Comment by carldebilly** [comment, marked as answer]
As outlined in the [official Uno Platform documentation](https://platform.uno/docs/articles/features/web-authentication-broker.html#usage-on-webassembly), the default _redirect uri_ is formatted as `<origin>/authentication-callback`, where the `<origin>` is automatically replaced by the browser's [`window.location.origin`](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/ts/Windows/Security/Authentication/Web/WebAuthenticationBroker.ts#L5) value.

You have the option to alter the _default value_ by modifying the [`WinRTFeatureConfiguration.WebAuthenticationBroker.DefaultCallbackPath`](https://github.com/unoplatform/uno/blob/master/src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.WebAuthenticationBroker.cs#L11) to a desired custom path.

The specific content hosted at this url is not crucial, as the URL will be intercepted as soon the window (or iframe) navigates to it. However, it's essential that is shares the same origina with the current running application. The [corresponding code can be found here](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/ts/Windows/Security/Authentication/Web/WebAuthenticationBroker.ts#L188-L191) (at least for the window version).

Some troubleshooting steps to consider on your end:
* If the issue pesists, debugging the JavaScript could shed light on the underlying problem.
* Utilize the F12 console to scout for any informative error messages.
* Verify that your external identity provider is accurately configured to acknowledge the new redirect URI, as misconfigurations there could also lead to errors.
* Utilize network monitoring tools within your browser's developer tools to observe the HTTP requests and responses. Look out for any failed requests, especially those related to authentication.
* Ensure that there are no Cross-Origin Resource Sharing (CORS) issues preventing the communication between your application and the identity provider.

**Reply by BertrandDeSaintBezier** [reply]
Thanks for the tip ! This ended up working by setting the [WinRTFeatureConfiguration.WebAuthenticationBroker.DefaultCallbackPath](https://github.com/unoplatform/uno/blob/master/src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.WebAuthenticationBroker.cs#L11) to match the callback we were getting on our end.
